src/core/reader/mod.rs

```diff
@@ -103,7 +161,81 @@ pub mod span {
         type Output = [u8];
 
         fn index(&self, index: Span) -> &'a Self::Output {
-            &self.full_contents[index.from..(index.from + index.len)]
+            &self.full_wasm_binary[index.from..(index.from + index.len)]
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use alloc::vec;
+
+    #[test]
+    fn move_start_to() {
+        let my_bytes = vec![0x11, 0x12, 0x13, 0x14, 0x15];
+        let mut wasm_reader = WasmReader::new(&my_bytes);
+
+        let span = Span::new(0, 0);
+        wasm_reader.move_start_to(span).unwrap();
+        // this actually dangerous, we did not validate there to be more than 0 bytes using the Span
+        wasm_reader.peek_u8().unwrap();
```

Comment:
Yep, it is a bit weird that the Span is used *actually* only for moving the pc forward with a number of bytes. Even though the Span also contains a length.

Reply:
FYI: The idea of having a `Span` object was just an idea of mine when I implemented the first code expression parsing.
It was used to mark where a code block starts and ends, but I think we might be able to refactor its API to be a little less obscure. Maybe we could also get rid of it completely🤷‍♂️

Reply:
For now we can keep the span's, they have some value in validation IIRC, because they allow to specify exactly the span of bytes for a given WASM item.